### ISSUE TITLE
Minor logic sensor additions

### ISSUE THREAD

**Issue by Xeloboyo** [Anuken/Mindustry-Suggestions#869]
following on from https://github.com/Anuken/Mindustry/issues/3048
_tldr: all blocks that don't use inventories have an inventory size of 10 and item input is just disabled._
Can there then be a sensor that detects whether a block will accept items then? I don't see any other way of directly distinguishing blocks from logic alone that allow items and those that dont.
Though having the blocks with disabled storages read 0 would be a more complete and intuitive solution.
 E.g. this quirk would confuse a processor that tells units to drop items in any unfull blocks it may be connected to if it's connected to like an on/off switch or something. The units will try to drop items into the switch because it reads a capacity of 10 and is holding no items.

While we're here the ability to sense the size of a linked block or detecting if a unit can build a block at some x,y would be a good addition to making unit building more applicable. I dont exactly see a practical way you can distinguish a unit failing to build an arbitrary block because its partially obstructed or because it ran out of resources and its just taking longer then usual; as getBlock will return `@air` in both cases. Having the size of the block will at least allow the method of knowing how many tiles to scan for `@solid` for any arbitrary linked block and a return variable that flags whether a unit has successfully begun building on `unit control build` would be a very convenient and intuitive method of detecting obstructions. Another method is having the construction-in-progress block be its own unique getblock return tag like `@air` and `@solid`.
